mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Puzzle.kt

```diff
@@ -0,0 +1,51 @@
+package ch.epfl.sdp.mobile.application.chess
+
+import ch.epfl.sdp.mobile.application.chess.engine.Color
+import ch.epfl.sdp.mobile.application.chess.engine.implementation.emptyBoard
+import ch.epfl.sdp.mobile.application.chess.engine.rules.Action
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation.BoardSnapshot
+
+/**
+ * Represents a complete [Puzzle] TODO: Will definitely change in the "Functional Puzzle Game"
+ * feature
+ */
+interface Puzzle {
+  /** The uid of the [Puzzle] */
+  val uid: String
+
+  /** The [BoardSnapshot] of the [Puzzle] */
+  val boardSnapshot: BoardSnapshot
+
+  /**
+   * The list of [Action] to be played by the "computer" and the player. The first action is the
+   * last moved played before the player starts the [Puzzle]
+   */
+  val puzzleMoves: List<Action>
+
+  /** The elo/rank (difficulty) of the [Puzzle] */
+  val elo: Int
+}
+
+/** Creates an empty [Puzzle]. */
+fun Puzzle(): Puzzle =
```

Comment:
[Suggestion] : change the name function name

```suggestion
fun EmptyPuzzle(): Puzzle =
```

Reply:
While I understand the idea, this is actually a common way to do this in Kotlin/Compose. For example, take a lool at `List()`.
Although different, you can think of it as an empty constructor.